generators/java/okhttp-gson/templates/libraries/okhttp-gson/ApiClient.mustache

```diff
@@ -1621,32 +1625,43 @@ public class ApiClient {
         return mpBuilder.build();
     }
 
+    {{! guessContentTypeFromFile dropped (part of FileTransfer now) }}
     /**
-     * Guess Content-Type header from the given file (defaults to "application/octet-stream").
+     * Add a Content-Disposition Header for the given key and file to the MultipartBody Builder.
      *
-     * @param file The given file
-     * @return The guessed Content-Type
+     * @param mpBuilder MultipartBody.Builder
+     * @param key The key of the Header element
+     * @param file The file to add to the Header
      */
-    public String guessContentTypeFromFile(File file) {
-        String contentType = URLConnection.guessContentTypeFromName(file.getName());
-        if (contentType == null) {
-            return "application/octet-stream";
-        } else {
-            return contentType;
+    {{! function updated to accept a FileTransfer object }}
+    private void addPartToMultiPartBuilder(MultipartBody.Builder mpBuilder, String key, FileTransfer file) {
+        Headers partHeaders = Headers.of("Content-Disposition", "form-data; name=\"" + key + "\"; filename=\"" + file.getFilename() + "\"");
+        MediaType mediaType = MediaType.parse(file.getContentType());
+
+        if ( file.getByteArray() != null ) {
```

Comment:
this **if** could be unnecessary if FileTransfer had a functino to return either the bytearray or file (`file.getContent()` or smth similar?

Reply:
In that case we should upcast to `Object` inside that function and then  downcast here to call the proper `addPart` function. Basically the code would remain quite the same, just a little bit longer. On the other hand, the FileTransfer object would be more complex. I'm not sure if we're getting any profit from this change... 🤔

Reply:
As discussed via PM, this can't be easily done since there isn't any common type [supported](https://square.github.io/okhttp/3.x/okhttp/okhttp3/RequestBody.html) by RequestBody.create() and we should therefore cast and call one or the other depending on the type.